projects/igniteui-angular/src/lib/grids/api.service.ts

```diff
@@ -155,8 +155,9 @@ export class GridBaseAPIService<T extends GridType> implements GridServiceType {
                 if (this.grid.pinnedRecords.find(r => r == cell.rowData)) {
```

Comment:
The pinnedRecords colelction is iterated two tiems to find the same item, can this be improved?

```
this.grid.pinnedRecords.find(r => r == cell.rowData)
this.grid.pinnedRecords.indexOf(cell.rowData);
```